### ISSUE TITLE
Accessing the underlying TLS stream

### ISSUE THREAD

**Issue by howardmoore** [tafia/hyper-proxy#13]
Would you consider making your ProxyStream implementation public and also switching form using tokio_tls to tokio_native_tls?

The reason I'm asking is because I'd like to be able to access the underlying TLS stream of a proxied connection, to do some further validation of the server certificate. This is almost possible using the Upgrade mechanism in hyper but to downcast the hyper::upgrade::Upgraded back to its underlying stream requires your ProxyStream struct to be made public and requires it to use a tokio_native_tls::TlsStream not a tokio_tls::TlsStream as the latter doesn't allow access to the underlying native_tls stream (and also looks to be deprecated in favour of the former).

**Comment by lynnux** [comment]
yes, same as tunnel, they both helpful, please consider make them pub

**Comment by tafia** [comment]
I would definitely consider it.
I am on a very low bandwidth since some time now but I'd be happy to share ownership with someone if needed.
